backend/app/Entities/User.php

```diff
@@ -43,16 +43,23 @@ public function sendPasswordResetNotification($token)
         $this->notify(new MailResetPasswordNotification($token));
     }
 
-    public function website()
+    public function websites()
     {
-        return $this->hasOne(Website::class, 'user_id');
+        return $this->belongsToMany(Website::class)->withPivot('role');
     }
 
-    public function getWebsite()
+    public function isWebsiteOwner(int $websiteId)
     {
-        return $this->website;
+        $userWebsite = auth()->user()
```

Comment:
why do you retrieve in model auth user? Check on the current user, or better all query to DB do inside the repository

Reply:
rename method to hasWebsite(int $websiteId), and check on current user, not authorized. If you need you can call it as auth()->user()->hasWebsite($websiteId), but not inside the model